lib/dependencies/ProvidedDependency.js

```diff
@@ -0,0 +1,70 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Florent Cailhol @ooflorent
+*/
+
+"use strict";
+
+const InitFragment = require("../InitFragment");
+const Template = require("../Template");
+const ModuleDependency = require("./ModuleDependency");
+
+class ProvidedDependency extends ModuleDependency {
+	constructor(request, originModule, name, specifier, range, loc) {
+		super(request);
+		this.originModule = originModule;
+		this.name = name;
+		this.specifier = specifier;
+		this.range = range;
+		this.loc = loc;
+	}
+
+	get type() {
+		return "provided";
+	}
+}
+
+function getImportVarName(dep) {
+	return `${Template.toIdentifier(dep.name)}__WEBPACK_PROVIDED_MODULE__`;
+}
+
+ProvidedDependency.Template = class ProvidedDependencyTemplate {
+	apply(dep, source, runtime) {
+		source.replace(
+			dep.range[0],
+			dep.range[1] - 1,
+			this.getContent(dep, runtime)
```

Comment:
Specifier missing

Reply:
I don't understand. The specifier is defined in `dep`. What are you referring to?